honeybee_grasshopper_core/src/HB Solve Adjacency.py

```diff
@@ -0,0 +1,129 @@
+# Honeybee: A Plugin for Environmental Analysis (GPL)
+# This file is part of Honeybee.
+#
+# Copyright (c) 2019, Ladybug Tools.
+# You should have received a copy of the GNU General Public License
+# along with Honeybee; If not, see <http://www.gnu.org/licenses/>.
+# 
+# @license GPL-3.0+ <http://spdx.org/licenses/GPL-3.0+>
+
+"""
+Solve adjacencies between a series of honeybee Rooms.
+_
+Note that rooms must have matching faces in order for them to be discovered as
+adjacent.
+-
+
+    Args:
+        _rooms: A list of honeybee Rooms for which adjacencies will be solved.
+        ep_constr_: Optional text for an energy construction to be used for all
+            adjacent Faces found in the process of solving adjacency. This text
+            will be used to look up a construction in the opaque construction library.
+            This can also be a custom OpaqueConstruction object. If no energy
+            construction is input here, the default will be assigned using the
+            Rooms' ConstructionSet.
+        ep_glz_constr_: Optional text for an energy construction to be used for all
```

Comment:
The workflow you are talking about with ConstructionSets already exists but you don't need to involve the "Solve Adjacency" component to make use of it.  You just have to make a ConstructionSet that has the interior constructions edited and assign that to all the zones that you want to have those constructions.  I'm putting together such a ConstructionSet component right now for honeybee-energy and I am building in some checks to give warnings for things like asymmetric interior walls of ConstructionSets or floors and ceilings not having reverse constructions. This should make it a lot easier if you want to do things like "make the entire interior of my building out of wood instead of concrete and gypsum."

However, the inputs on the "Solve Adjacency" component here are important for when you have exceptions to these broad rules. I am thinking for cases where you model a DSF or a a Trombe wall cavity as a separate zone and you need to make sure that just the interior wall of the Trombe wall is a thermally massive construction.